src/main/scala/dotty/dokka/tasty/TypesSupport.scala

```diff
@@ -0,0 +1,125 @@
+package dotty.dokka.tasty
+
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.model.{Projection => JProjection}
+import collection.JavaConverters._
+
+trait TypesSupport:
+    self: TastyParser =>
+    import reflect._
+
+    extension on (tpeTree: Tree):
+        def dokkaType(using cxt: reflect.Context): Bound = 
+            val data = tpeTree match
+                case TypeBoundsTree(low, high) => 
+                    typeBound(low.tpe, low = true) ++ typeBound(high.tpe, low = false)
+                case tpeTree: TypeTree => inner(tpeTree.tpe)
+                case term:  Term => inner(term.tpe)
+            
+            new TypeConstructor(tpeTree.symbol.dri, data.asJava, FunctionModifiers.NONE)
+      
+    private def text(str: String): JProjection = new UnresolvedBound(str)
+    
+    private def texts(str: String): List[JProjection] = List(text(str))
+    
+    private def noSupported(name: String): List[JProjection] =
+        println(s"WARN: Unsupported type: $name") 
+        List(text(s"Unsupported[$name]"))
+    
+    private def link(symbol: reflect.Symbol)(using cxt: reflect.Context): JProjection = new OtherParameter(symbol.dri, symbol.name)
+    
+    private def commas(lists: List[List[JProjection]]) = lists match
+        case List(single) => single
+        case other => other.reduce((r, e) => r ++ texts(", ") ++ e)
+    
+    // TODO add support for all types signatures that makes sense
+    private def inner(tp: reflect.TypeOrBounds)(using cxt: reflect.Context): List[JProjection] = tp match
+        case OrType(left, right) => inner(left) ++ texts(" | ") ++ inner(right)
+        case AndType(left, right) => inner(left) ++ texts(" & ") ++ inner(right)
+        case ByNameType(tpe) => text("=> ") :: inner(tpe)
+        case ConstantType(constant) => texts(constant.value.toString) 
+        case ThisType(tpe) => inner(tpe)
+        case AnnotatedType(tpe, _) => inner(tpe)
+        // case TypeLambda(paramNames, paramTypes, resType) => ConstantReference(tp.show) //TOFIX
+        case Refinement(parent, name, info) =>
+            // val tuple = convertTypeOrBoundsToReference(reflect)(info) match {
+            //     case r if (info match {case info: TypeBounds => true case _ => false}) => ("type", name, r)
+            //     case r@TypeReference(_, _, _, _) => ("val", name, r)
+            //     case ByNameReference(rChild) => ("def", name, rChild)
+            //     case r => throw new Exception("Match error in info of Refinement. This should not happen, please open an issue. " + r)
+            // }
+            // convertTypeToReference(reflect)(parent) match {
+            //     case RefinedReference(p, ls) =>
+            //     RefinedReference(p, ls:+tuple)
+            //     case t => RefinedReference(t, List(tuple))
+            // }
+            noSupported("Refinement") 
```

Comment:
Refinements are a bit complicated - let me know if you could use a description of what they mean.